cmake/Modules/FindLIBNET.cmake

```diff
@@ -41,27 +41,43 @@ FIND_PROGRAM(LIBNET_CONFIG libnet-config)
 add_library(libnet INTERFACE)
 
 IF (LIBNET_CONFIG)
-  EXEC_PROGRAM(${LIBNET_CONFIG} ARGS --libs OUTPUT_VARIABLE _LIBNET_LIBRARIES)
+  EXEC_PROGRAM(${LIBNET_CONFIG} ARGS --defines OUTPUT_VARIABLE _LIBNET_DEFINES)
+  #
+  # this is due to libnet-config provides old fashined defines, which triggers warning on newer systems
+  # for details see: https://github.com/libnet/libnet/pull/71
+  #
+  string(APPEND _LIBNET_DEFINES " -D_DEFAULT_SOURCE")
+  target_compile_definitions(libnet INTERFACE ${_LIBNET_DEFINES})
+
   EXEC_PROGRAM(${LIBNET_CONFIG} ARGS --cflags OUTPUT_VARIABLE _LIBNET_CFLAGS)
-  string(REGEX REPLACE "[\r\n]" " " _LIBNET_LIBRARIES "${_LIBNET_LIBRARIES}")
+  EXEC_PROGRAM(${LIBNET_CONFIG} ARGS --libs OUTPUT_VARIABLE _LIBNET_LIBRARIES)
   string(REGEX REPLACE "[\r\n]" " " _LIBNET_CFLAGS "${_LIBNET_CFLAGS}")
-  set (LIBNET_LIBRARIES ${_LIBNET_LIBRARIES} CACHE STRING "The libraries needed for LIBNET")
-  set (LIBNET_CFLAGS ${_LIBNET_CFLAGS} CACHE STRING "The compiler switches needed for LIBNET")
-  set (LIBNET_FOUND TRUE CACHE BOOL "LibNet is found")
+  string(REGEX REPLACE "[\r\n]" " " _LIBNET_LIBRARIES "${_LIBNET_LIBRARIES}")
 
-# this is due to libnet-config provides old fashined defines, which triggers warning on newer systems
-# for details see: https://github.com/libnet/libnet/pull/71
+  string(REPLACE "-I" "" _LIBNET_INCLUDES "${_LIBNET_CFLAGS}")
```

Comment:
I think we should not modify the cflags queried from libnet-config, unless we absolutely need to.
CMake accepts `-I` as well.